src/facade/dragonfly_connection.cc

```diff
@@ -101,21 +122,92 @@ struct Connection::Shutdown {
   }
 };
 
+// Used as custom deleter for Request object
+struct Connection::RequestDeleter {
+  void operator()(Request* req) const {
+    mi_free(req);
```

Comment:
i like the idea of custom delete but i am pretty sure you have a memory leak here.

Reply:
I checked for memory leaks. I didn't find any. any pointer that was allocated, was released as well (or at least the call to released it was called, if mi_free don't really free it, then we would have a memory leak here), but this is just the same as the old code. There is a single allocation of memory to the req itself. I can add an explicit call to dtor, but its trivial.

Reply:
it's not trivial at all. it has non-pod members that by themselves take memory.  for example `storage` member is leaking for requests that take more than 120 bytes. If I am correct, it could be easily proven by sending a pipelined traffic to the server.

Reply:
This is why I added the explicit call to the dtor in the code. So that if we have some object with none trivial dtor it would be called. Right now the code does the same as it did before, only we don't need to call it explicitly since it is done in the deleter of the unique ptr